### ISSUE TITLE
Improved shortcuts for inserting headings

### ISSUE THREAD

**Issue by Clorith** [WordPress/gutenberg#7438]
Currently the `/heading` shortcut adds a H2 by default, but the flow for creating other headings is cumbersome, you either have to start using your mouse, or jump back and forth using tabs.

For a good direct flow (for those who like commands over hashtags) would be to keep typing, `/heading3` (a bit wordy/convoluted) or `/h3` (quick and easy) for example, although this would mean we need some way to introduce multiple shortcuts per block with the ability to tie them to attributes of that block, possibly a bit much for the first iteration, but definitely enhancements I'd love to see moving forward.

**Comment by danielbachhuber** [comment]
`/h3` would be pretty neat.

**Comment by TimothyBJacobs** [comment]
The markdown `###` shortcuts work for this, but they aren't very discoverable.

**Comment by Clorith** [comment]
Exactly, they're not and once you start with the flow of using slash-commands it's awkward to jump back to other shortcuts from experience, it just feels natural to start off a new line with a "command over a secret string".

**Comment by bph** [comment]
Thank you for picking this up @Clorith  It's a great feature in the current editor not to have to change controls to a mouse to convert a paragraph to a heading. The shortcuts would make it par with the old editor and help with the transition, as people won't expect that 'their' keyboard short
![screen shot 2018-06-28 at 9 14 20 am](https://user-images.githubusercontent.com/39980/42036542-cdf82c72-7ab3-11e8-86cc-582b2d9a4a9f.png)
cuts don't work anymore.

**Comment by talldan** [comment]
Noting that there's an old PR that adds some shortcuts for headings. Not sure if that's something we can get rolling again as it's fairly out-of-date. I may have a look at it later this week or next week if I can.

https://github.com/WordPress/gutenberg/pull/4436

**Comment by uncertainquark** [comment]
The keyboard shortcuts for assigning heading tags should preferably be the same as other document editor i.e `Ctrl+Alt+1/2/3` or `Alt+1/2/3`.

**Comment by coollorenzo** [comment]
Would love to see CTRL+ALT+2 to convert to h2!

**Comment by phpbits** [comment]
These shortcuts are available via [EditorsKit](https://wordpress.org/plugins/block-options/) plugin:

<img width="636" alt="Screen Shot 2020-06-03 at 6 00 55 PM" src="https://user-images.githubusercontent.com/3365507/83623644-2cae1280-a5c4-11ea-83c0-433283af60c7.png">

**Comment by uncertainquark** [comment]
It’s been really long and yet we can’t change Headings with keyboard shortcuts. I’ve stopped writing articles in the Block editor entirely. It’s great for creating complex pages and layouts but terrible for writing. Even inserting images is such a pain.
![telegram-cloud-photo-size-5-6122697952188280492-y](https://user-images.githubusercontent.com/22585732/97956323-5adf8a80-1dce-11eb-8657-1b6cc0b6e799.jpg)

**Comment by talldan** [comment]
Fixed by https://github.com/WordPress/gutenberg/pull/26597.

**Comment by uncertainquark** [comment]
In which update will this change for Headings land? The linked PR shows 9.4 but that’s already out?

**Comment by talldan** [comment]
@uncertainquark 9.4 isn't out yet (the current plugin version is 9.2.2—https://wordpress.org/plugins/gutenberg/).

Unfortunately this change was merged just to late for 9.3, which will be release tomorrow. It'll be in 9.4, which should be released on the 18th November.

**Comment by uncertainquark** [comment]
Nov 18 isn’t too late. :)
I'll get the update automatically, right? I’m using Wordpress.com.
